addon/controllers/building/edit/dwelling/edit.js

```diff
@@ -0,0 +1,78 @@
+import Controller from "@ember/controller";
+import { inject as service } from "@ember/service";
+import { task, lastValue } from "ember-concurrency-decorators";
+import Models from "ember-ebau-gwr/models";
+import DwellingValidations from "ember-ebau-gwr/validations/dwelling";
+
+export default class BuildingEditDwellingEditController extends Controller {
+  Models = Models;
+  DwellingValidations = DwellingValidations;
+
+  @service("dwelling") dwellingAPI;
+  @service building;
+  @service intl;
+  @service notification;
+
+  @lastValue("fetchDwelling") dwelling;
+  @task
+  *fetchDwelling() {
+    try {
+      yield this.fetchEntrances.perform();
+      if (this.model.dwelling?.isNew) {
+        this.model.dwelling.EDID = this.entrances[0].EDID;
+        return this.model.dwelling;
+      }
+
+      const { EDID, dwelling } = yield this.dwellingAPI.getFromCacheOrApi(
+        this.model.dwellingId,
+        this.model.buildingId
+      );
+      dwelling.oldEDID = EDID;
+      return dwelling;
+    } catch (error) {
+      console.error(error);
+      this.notification.danger(this.intl.t("ember-gwr.dwelling.loadingError"));
+    }
+  }
+
+  @lastValue("fetchEntrances") entrances;
+  @task
+  *fetchEntrances() {
+    try {
+      const building = yield this.building.getFromCacheOrApi(
+        this.model.buildingId
+      );
+      return building.buildingEntrance;
+    } catch (error) {
+      console.error(error);
+      this.notification.danger(
+        this.intl.t("ember-gwr.building.entrances.error")
+      );
+    }
+  }
+
+  @task
```

Comment:
Should this maybe be a `@dropTask` to prevent duplicate creations or is that otherwise prevented?

Reply:
Done. Also on all the other mutate tasks

Reply:
:ok_hand: